File: src/main/java/vip/mystery0/base/springboot/utils/rest/JSON.java
package vip.mystery0.base.springboot.utils.rest;

import com.fasterxml.jackson.core.type.TypeReference;

/**
 * @author Mystery0
 */
public interface JSON {
    <T> T fromJson(String json, Class<T> clazz);

    <T> T fromJson(String json, TypeReference<T> typeReference);

    <T> String toJson(T obj);
}

File: src/main/java/vip/mystery0/base/springboot/utils/rest/JSON.kt
package vip.mystery0.base.springboot.utils.rest

import java.lang.reflect.Type

/**
 * @author Mystery0
 */
interface JSON {
    fun <T> fromJson(json: String, clazz: Class<T>): T
    fun <T> fromJson(json: String, type: Type): T
    fun <T> toJson(obj: T): String
}